internal/mq/msgstream/mqwrapper/pulsar/pulsar_consumer.go

```diff
@@ -145,6 +146,24 @@ func (pc *Consumer) GetLatestMsgID() (mqwrapper.MessageID, error) {
 	return &pulsarID{messageID: msgID}, err
 }
 
+func (pc *Consumer) CheckPreTopicValid(topic string) error {
+	latestMsgID, err := pc.GetLatestMsgID()
+	log := log.With(zap.String("channel", topic))
+	// Pulsar creates that topic under the namespace provided in the topic name automatically
+	if err != nil {
+		log.Error("pulsar get latest msg ID failed", zap.Error(err))
+		return err
+	}
+
+	if latestMsgID.AtEarliestPosition() {
```

Comment:
just keep the err branch with:
if !latestMsgID.AtEarliestPosition() { ... }
and move the line 159 out